### PR TITLE
MCMC tests: check if inj params within mean+-2std and within 90%

### LINKED ISSUE #112: MCMC tests randomly fail my (vaguely guessed) tolerance thresholds
The new MCMC tests from #113 ran fine the first few times, but actually fail unreasonably often when the parameter recovery happens to be randomly off-tolerance. Since, to keep the test runtime bounded, the runs are kept very short and unlikely to be perfectly converged even for the tiny test boxes, this shouldn't have come as a surprise to me...

I'm afraid the clean solution will be to test if mean+-stdev (or 90% intervals) cover the injection value, instead of the maxF point being within some guessed tolerance.

### PR DESCRIPTION
**Pull request by dbkeitel**
resolves #112 (I hope)

Still cannot be **guaranteed** to always succeed, but should be much more reliable than fixed tolerances.

If it fails again more often than I'd hope, we can try fixing "nice" random seeds.

### CODE REVIEW

**Comment by dbkeitel** [comment]
Tried a few hacks to get specifically the most failing-prone cases safer, but it all wasn't enough.

Now:

- only check injection parameters are within mean+-3std and within 99%, which for the narrow test case priors isn't really impressive at all, but better than nothing
- use a fixed random seed for data generation; this way the only run-to-run randomness is in the MCMC itself, which is much less than that in noise realisations

So it still cannot be guaranteed to always work, but after a lot of false starts I hope it is now reasonably robust...

File: pyfstat/mcmc_based_searches.py [hunk 1]
```diff
@@ -1804,9 +1804,15 @@ def get_summary_stats(self):
...
-            )
+            (
+                d[k]["lower99"],
+                d[k]["lower90"],
+                d[k]["lower50"],
+                d[k]["median"],
+                d[k]["upper50"],
+                d[k]["upper90"],
+                d[k]["upper99"],
+            ) = np.quantile(s, [0.005, 0.05, 0.25, 0.5, 0.75, 0.95, 0.995])
```
Comment (Rodrigo-Tenorio) [inline pyfstat/mcmc_based_searches.py:1815]: I'm not sure if I get this: Why e.g. `lower99` goes with 0.005?

File: tests.py [hunk 2]
```diff
@@ -723,7 +723,6 @@ class MCMCSearch(Test):
     label = "TestMCMCSearch"
 
     def test_fully_coherent_MCMC(self):
-
```
Comment (Rodrigo-Tenorio) [inline tests.py:726]: Let me have a quick look on this

File: pyfstat/mcmc_based_searches.py (hunk 1 above)
Comment (dbkeitel) [reply pyfstat/mcmc_based_searches.py:1815]: symmetric 99% interval = 99% contained around the median -> [0.005,0.995] with half a percent left out either side

Comment (Rodrigo-Tenorio) [reply pyfstat/mcmc_based_searches.py:1815]: Of course, that makes sense.

File: tests.py [hunk 3]
```diff
@@ -67,6 +67,53 @@ def tearDownClass(self):
                 logging.warning("{} not removed prior to tests".format(self.outdir))
 
 
+class MCMCSearchTest(Test):
```
Comment (Rodrigo-Tenorio) [inline tests.py:70]: @dbkeitel opinions?

Comment (dbkeitel) [reply tests.py:70]: Yes, good idea! The whole test suite could be cleaned up with more inheritance, but the again, it seems it should be migrated from an unholy unittest-pytest mix to pure pytest later, so it's not worth much more effort at this point I'd say.

Could you move this to further down though, just before the first child class (`class MCMCSearch(MCMCSearchTest)`)?

Comment (Rodrigo-Tenorio) [reply tests.py]: Done

**Pull request merged**